kernel/boot_info/src/lib.rs

```diff
@@ -7,7 +7,10 @@
 #![feature(type_alias_impl_trait)]
 #![no_std]
 
+#[cfg(feature = "multiboot2")]
 pub mod multiboot2;
+#[cfg(feature = "uefi")]
+pub mod uefi;
```

Comment:
not necessary, but maybe a small quality of life improvement: shall we throw a compile error if neither multiboot2 nor uefi feature is activated?

Reply:
No, because consumer crates don't depend on either feature e.g. `memory`